pkg/operators/localmanager/localmanager.go

```diff
@@ -342,99 +342,103 @@ func (l *localManagerTrace) PreGadgetRun() error {
 		},
 	}
 
-	// If --host is set, we do not want to create the below map because we do not
-	// want any filtering.
-	if setter, ok := l.gadgetInstance.(MountNsMapSetter); ok {
-		if !host {
-			if l.manager.igManager == nil {
-				return fmt.Errorf("container-collection isn't available")
-			}
+	if l.gadgetInstance != nil {
```

Comment:
nit: use an early return? I also think the operations above (to the the contianerSelector and so on), should be avoided if this is nil, they aren't used in that case.

Reply:
I'll extract that to a new function - an early return might confuse if stuff is added at the end of the function otherwise.